openmetadata-ui/src/main/resources/ui/src/utils/executionUtils.tsx

```diff
@@ -106,3 +111,125 @@ export const getStatusLabel = (status: string) => {
       return;
   }
 };
+
+export const getExecutionElementByKey = (
+  key: string,
+  viewElements: {
+    key: string;
+    value: ReactNode;
+  }[]
+) => viewElements.find((v) => v.key === key);
+
+// check if current task is downstream task of other tasks
+const checkIsDownStreamTask = (currentTask: Task, tasks: Task[]) =>
+  tasks.some((taskData) =>
+    taskData.downstreamTasks?.includes(currentTask.name)
+  );
+
+export const getTreeData = (
+  tasks: Task[],
+  viewData: Record<string, ViewDataInterface[]>,
+  isLabelList = false
```

Comment:
For this you are calling this method twice, Can we improve it? Calling this method once only and might handle it in rendering?